src/Microsoft.AspNetCore.Authentication.OpenIdConnect/Events/IOpenIdConnectEvents.cs

```diff
@@ -38,17 +28,22 @@ public interface IOpenIdConnectEvents : IRemoteAuthenticationEvents
         /// <summary>
         /// Invoked before redirecting to the identity provider to authenticate.
         /// </summary>
-        Task RedirectToAuthenticationEndpoint(RedirectContext context);
+        Task RedirectToIdentityProvider(RedirectContext context);
```

Comment:
Heh, I've never said the events model was specific to AAD :open_mouth: 

I just said that `OpenIdConnectOptions` already has AAD-specific options like `Resource` and that I'm much more comfortable with adding new first-class options than forcing folks to use the events approach, which requires more work and is more complicated for beginners.

Anyway, it doesn't change anything to the fact I believe `RedirectToIdentityProvider` is a poor name.